.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,256 @@
+name: Build and test
+
+on:
+  push:
+    branches:
+      - master
+      - "*-rc"
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - "**"
+
+jobs:
+  verify_files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y gnupg2
+
+      - name: Verify files
+        run: |
+          curl -sSL https://secchannel.rsk.co/SUPPORT.asc | gpg --import -
+          gpg --verify SHA256SUMS.asc && sha256sum --check SHA256SUMS.asc
+
+  clone_rskj_repo:
+    needs: verify_files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout RSKj repo
+        uses: actions/checkout@v4
+        with:
+          repository: rsksmart/rskj
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: rskj
+          fetch-depth: 0
+      
+      - name: Determine branch to checkout for PR
+        if: github.event_name == 'pull_request'
+        working-directory: rskj
+        env:
+          PR_BRANCH: ${{ github.head_ref }}
+        run: |
+          IS_RSKJ_BRANCH=`git ls-remote --heads origin $PR_BRANCH`
+          if test -n "${IS_RSKJ_BRANCH}"; then
+            echo "Found matching branch name in RSKj repo"
+            CHECKOUT_REF="$PR_BRANCH"
+          else
+            echo "Using master for RSKj"
+            CHECKOUT_REF="master"
+          fi
+          echo "CHECKOUT_REF=$CHECKOUT_REF" >> $GITHUB_ENV
+          
+      - name: Determine branch to checkout for push 
+        if: github.event_name != 'pull_request'
+        working-directory: rskj
+        env:
+          POW_REF: ${{ github.ref }}
+        run: |
+          IS_RSKJ_REF=`git ls-remote --refs $POW_REF`
+          if test -n "${IS_RSKJ_REF}"; then
+            echo "Found matching ref in RSKj"
+            CHECKOUT_REF="$POW_REF"
+          else
+            echo "Using master for RSKj"
+            CHECKOUT_REF="master"
+          fi
+          echo "CHECKOUT_REF=$CHECKOUT_REF" >> $GITHUB_ENV
+
+      - name: Check out appropriate rskj reference
+        working-directory: rskj
+        run: |
+          git switch "${{ env.CHECKOUT_REF }}"
+
+      - name: Persist RSKJ
+        uses: actions/upload-artifact@v4
+        with:
+          name: rskj
+          path: rskj
+  
+  build_federator_node:
+    runs-on: ubuntu-latest
+    container: openjdk:8-jdk
```

Comment:
could we use java17 instead? like in [here](https://github.com/rsksmart/rskj/pull/2624/files#diff-48c0ee97c53013d18d6bbae44648f7fab9af2e0bf5b0dc1ca761e18ec5c478f2R14-R23)?

Reply:
Addressed. Switched to Java 17